File: trunk/src/robotcore/Robot.cpp
/*
 * Robot.cpp
 * RobotQt - Robot Simulation
 *
 * Created by Felipe Tonello on 2008-12-10.
 * Copyright 2008 Felipe Ferreri Tonello. All rights reserved.
 * http://robotqt.googlecode.com/
 *
 * Revision: $Rev$
 * Author: $Author$
 * Date: $Date$
 */

#include "Robot.h"


Robot::Robot(QString _name, quint8 _id) :
    name(_name), id(_id)
{
    setWheelSize(100); // 100 cm = 1 m
}

Robot::~Robot()
{
    //TODO: Delete all Motor and Sensor
}

QString Robot::getName() const
{
    return name;
}

quint8 Robot::getId() const
{
    return id;
}

quint8 Robot::getWheelSize() const
{
    return wheelSize;
}

void Robot::setWheelSize(quint8 cm)
{
    wheelSize = cm;
}

void Robot::set(Motor *newMotor)
{
    motorList.append(newMotor);
}

void Robot::set(Sensor *newSensor)
{
    sensorList.append(newSensor);
}

Motor * Robot::getMotor(quint8 _id) const
{
    for (int i = 0; i < motorList.size(); i++) {
        if (motorList[i].getId() == _id) {
            return motorList[i];
        }
    }
    return NULL;
}

Sensor * Robot::getSensor(quint8 _id) const
{
    for (int i = 0; i < sensorList.size(); i++) {
        if (sensorList[i].getId() == _id) {
            return sensorList[i];
        }
    }
    return NULL;
}

File: pythonsrc/robotcore/robot.py
"""
robot.py
RobotQt - Robot Simulation

Created by Felipe Tonello on 2009-28-01.
Copyright 2009 Felipe Ferreri Tonello. All rights reserved.
http://robotqt.googlecode.com/

Revision: $Rev$
Author: $Author$
Date: $Date$
"""

from PyQt4 import QtCore
from Motor import Motor
from Sensor import Sensor

# TODO: debug message

class Robot(object):
    def __init__(self, name)
        self.name = QtCore.QString(name)
        self.stuff_list = []

    def set(self, stuff):
        if isinstance(stuff, Motor) or isinstance(stuff, Sensor)
            # If already exists or not
            if not _exists(stuff):
                self.stuff_list.append(stuff)
                return True
            else:
                raise Exception("stuff(%s) is not a Motor nor a Sensor" %
                stuff.__str__)
        else:
            # stuff is not a Motor nor a Sensor..
            raise Exception("stuff(%s) is not a Motor nor a Sensor" %
            stuff.__str__)

    def get(self, name):
        for stuff in self.stuff_list:
            if stuff.name == name:
                return stuff
        raise Exception("There is no Motor nor Sensor with name = %s" % name)

    def _exists(self, stuff):
        exists = False
        for s in self.stuff_list:
            if s.name == stuff.name:
                exists = True
                break
        return exists
